### PR TITLE
Add agentic multimodal evaluation fixtures

### LINKED ISSUE #716: OpenSearch-VL alignment: add agentic multimodal evaluation suites - Milestone 1: define suite taxonomy and fixtures - Unit 1: Add fixture packages for image-grounded multi-hop QA, visual retrieval QA, and document lookup QA.
## Executable Unit

Add fixture packages for image-grounded multi-hop QA, visual retrieval QA, and document lookup QA.

## Parent Context

- Direction: `OpenSearch-VL alignment: add agentic multimodal evaluation suites`
- Plan/Milestone: `Milestone 1: define suite taxonomy and fixtures`

## Definition Of Done

- The change is scoped to the smallest useful implementation slice.
- The governing spec or plan is created or updated before broad code changes.
- Tests cover the new behavior or contract edge case.
- Metrics and verification output are recorded in the PR evidence.

## Suggested Verification

- `git diff --check`
- Focused pytest or Swift test for the touched code path
- Changed-scope coverage for modified Python or Swift files when measurable
- Any relevant benchmark/evaluation artifact if this changes measured behavior

### PR DESCRIPTION
**Pull request by Zigfreidish**
## Summary
- Add three repository-owned agentic multimodal development fixture packages for image-grounded multi-hop QA, visual retrieval QA, and document lookup QA.
- Add a deterministic static contract test that validates manifests, sample rows, package-local assets, allowed tool coverage, evidence ids, and replay through the unified agentic tool runtime.
- Update the benchmark/evaluation contract to name the new development fixtures and clarify that they are contract/regression fixtures, not benchmark scorecards.

## Plan or Spec
- `docs/plans/2026-05-21-agentic-multimodal-evaluation-fixtures.md`
- `docs/benchmark-evaluation-contract.md`
- Closes #716

## Commands Run
```text
PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python pytest -q services/mlx-worker-python/tests/test_agentic_multimodal_evaluation_fixtures.py
# RED: expected 3 failures because the fixture package directories did not exist yet.

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python pytest -q services/mlx-worker-python/tests/test_agentic_multimodal_evaluation_fixtures.py
# PASS: 3 passed.

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python coverage run -m pytest -q services/mlx-worker-python/tests/test_agentic_multimodal_evaluation_fixtures.py
# PASS: 3 passed.

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python coverage json -o .runtime/coverage-agentic-multimodal-fixtures.json
# PASS: coverage JSON written.

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python coverage report -m services/mlx-worker-python/tests/test_agentic_multimodal_evaluation_fixtures.py
# PASS: 100% statement coverage for the new fixture contract test.

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python python scripts/python_changed_line_coverage.py --coverage-json .runtime/coverage-agentic-multimodal-fixtures.json --diff-from origin/main services/mlx-worker-python/tests/test_agentic_multimodal_evaluation_fixtures.py
# PASS: TOTAL 100.00% (72/72).

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python python scripts/pr_scoped_performance_scope.py --registry infra/perf/pr_scoped_probes.json --changed-files-json .runtime/pr-scope/changed-files.json --output .runtime/pr-scope/scope.json
# PASS: selected_count=0, selected_probes=[].

python3 -m json.tool services/mlx-worker-python/fixtures/evaluation/agentic-multihop-qa.dev.v1/manifest.json >/dev/null
python3 -m json.tool services/mlx-worker-python/fixtures/evaluation/agentic-visual-retrieval-qa.dev.v1/manifest.json >/dev/null
python3 -m json.tool services/mlx-worker-python/fixtures/evaluation/agentic-document-lookup-qa.dev.v1/manifest.json >/dev/null
# PASS: all manifests parse as JSON.

git diff --check
git diff --cached --check
# PASS: no whitespace errors.

git commit -m "test: add agentic multimodal evaluation fixtures"
# PASS: pre-commit completed make swift-test, make py-test, make integration-test, and PR scoped performance report.
# make py-test: 2930 passed, 14 skipped.
# make integration-test: 114 passed, 1 skipped.
# PR scoped performance: Status ok; selected probes 0; regressions 0; verification failures 0.

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python pytest -q services/mlx-worker-python/tests/test_agentic_multimodal_evaluation_fixtures.py
# PASS after review fix: 3 passed.

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python coverage run -m pytest -q services/mlx-worker-python/tests/test_agentic_multimodal_evaluation_fixtures.py
PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python coverage json -o .runtime/coverage-agentic-multimodal-fixtures-review.json
PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python python scripts/python_changed_line_coverage.py --coverage-json .runtime/coverage-agentic-multimodal-fixtures-review.json --diff-from origin/main services/mlx-worker-python/tests/test_agentic_multimodal_evaluation_fixtures.py
git diff --check
# PASS after review fix: 3 passed; changed-line coverage TOTAL 100.00% (72/72); no whitespace errors.

git commit -m "test: tighten multimodal fixture observation assertions"
# PASS: pre-commit completed make swift-test, make py-test, make integration-test, and PR scoped performance report.
# make py-test: 2930 passed, 14 skipped.
# make integration-test: 114 passed, 1 skipped.
# PR scoped performance: Status ok; changed files 1; selected probes 0; regressions 0; verification failures 0.

git merge origin/main
# PASS: merged origin/main at b47f816f with no conflicts.

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python pytest -q services/mlx-worker-python/tests/test_agentic_multimodal_evaluation_fixtures.py services/mlx-worker-python/tests/test_tool_registry.py services/mlx-worker-python/tests/test_pr_scoped_performance.py
# PASS after origin/main merge: 287 passed.

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python coverage run -m pytest -q services/mlx-worker-python/tests/test_agentic_multimodal_evaluation_fixtures.py
PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python coverage json -o .runtime/coverage-agentic-multimodal-fixtures-merged.json
PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python python scripts/python_changed_line_coverage.py --coverage-json .runtime/coverage-agentic-multimodal-fixtures-merged.json --diff-from origin/main services/mlx-worker-python/tests/test_agentic_multimodal_evaluation_fixtures.py
PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python python scripts/pr_scoped_performance_scope.py --registry infra/perf/pr_scoped_probes.json --changed-files-json .runtime/pr-scope-merged/changed-files.json --output .runtime/pr-scope-merged/scope.json
git diff --check
# PASS after origin/main merge: 3 passed; changed-line coverage TOTAL 100.00% (72/72); selected_count=0; no whitespace errors.
```

## Coverage and Metrics
- Changed-line coverage: 100.00% (72/72) for `services/mlx-worker-python/tests/test_agentic_multimodal_evaluation_fixtures.py`.
- Focused fixture contract coverage: 100% statement coverage for the new test module.
- Deterministic replay metric: all package tool calls complete with `failed_count=0` and `timeout_count=0` as asserted by the focused test.
- Runtime performance metrics: N/A because this change adds static fixture packages, documentation, and a replay contract test only. The PR scoped performance report selected 0 registered probes and reported 0 regressions, including after merging the latest `origin/main`.
- Observability mode: off for production runtime paths; the only evidence path is deterministic fixture replay in tests.
- Probe overhead: N/A because no runtime probes or production instrumentation were added.

## Known Gaps
- Runner/scorer behavior is intentionally unchanged.
- Formal sample field taxonomy remains deferred to #717.
- Trajectory and judge report artifacts remain deferred to later milestone units.

## Evidence Checklist
- [x] Relevant plan or spec is identified.
- [x] Behavior changes are reflected in the relevant docs.
- [x] Protocol changes include regenerated generated artifacts. N/A: no protocol changes.
- [x] Dependency changes include updated lockfiles. N/A: no dependency changes.
- [x] Relevant tests were run.
- [x] A metrics report is included, or `N/A` is stated explicitly with the reason.
- [x] Observability mode, probe overhead, and any deferred debug or evidence work are recorded, or `N/A` is stated explicitly with the reason.
- [x] Deferred work and known gaps are stated explicitly.

### CODE REVIEW

File: services/mlx-worker-python/tests/test_agentic_multimodal_evaluation_fixtures.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            )
+            evidence = json.dumps(run.to_sample_evidence(), sort_keys=True)
+
+            assert run.registry_receipt["toolset_version"] == "melix.agentic_tools.builtin.v1"
+            assert run.metrics["agentic_tool.call_count"] == float(len(sample["tool_calls"]))
+            assert run.metrics["agentic_tool.failed_count"] == 0.0
+            assert run.metrics["agentic_tool.timeout_count"] == 0.0
+            assert run.metrics["agentic_tool.completed_count"] == float(len(sample["tool_calls"]))
+            for token in str(sample["expected_answer"]).split():
+                assert token in evidence, f"{package_id} replay evidence missing answer token {token!r}"
```
[the remark this answers is not in the document]
Comment (Zigfreidish) [reply services/mlx-worker-python/tests/test_agentic_multimodal_evaluation_fixtures.py]: Fixed in 415c2af3. The replay assertion now serializes only `run.observations`, so expected-answer tokens must come from tool observations rather than the request/tool-call payload.
